modules/function/src/main/java/com/opengamma/strata/function/calculation/swaption/SwaptionMarketLookup.java

```diff
@@ -0,0 +1,144 @@
+   * 
+   * @return the type of the parameter implementation
+   */
+  @Override
+  default Class<? extends CalculationParameter> queryType() {
+    return SwaptionMarketLookup.class;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the set of indices that volatilities are provided for.
+   *
+   * @return the set of indices
+   */
+  public abstract ImmutableSet<IborIndex> getVolatilityIndices();
+
+  /**
+   * Gets the identifiers used to obtain the volatilities for the specified currency.
+   * <p>
+   * The result will typically refer to a surface or cube.
+   * If the index is not found, an exception is thrown.
+   *
+   * @param index  the index for which a discount curve is required
+   * @return the set of market data identifiers 
+   * @throws IllegalArgumentException if the index is not found
+   */
+  public abstract ImmutableSet<MarketDataKey<?>> getVolatilityIds(IborIndex index);
+
+  //-------------------------------------------------------------------------
+  /**
+   * Creates market data requirements for the specified indices.
+   * 
+   * @param indices  the indices, for which forward curves and time-series will be needed
+   * @return the requirements
+   */
+  public abstract FunctionRequirements requirements(Set<IborIndex> indices);
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains a filtered view of the complete set of market data.
+   * <p>
+   * This method returns an instance that binds the lookup to the market data.
+   * The input is {@link CalculationMarketData}, which contains market data for all scenarios.
+   * 
+   * @param marketData  the complete set of market data for all scenarios
+   * @return the filtered market data
+   */
+  public default SwaptionCalculationMarketView marketView(CalculationMarketData marketData) {
+    return DefaultSwaptionCalculationMarketView.of(this, marketData);
+  }
+
+  /**
+   * Obtains a filtered view of the complete set of market data.
+   * <p>
+   * This method returns an instance that binds the lookup to the market data.
+   * The input is {@link CalculationMarketData}, which contains market data for one scenario.
+   * 
+   * @param marketData  the complete set of market data for one scenario
+   * @return the filtered market data
+   */
+  public default SwaptionMarketView marketView(MarketData marketData) {
+    return DefaultSwaptionMarketView.of(this, marketData);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains swaption volatilities based on the specified market data.
+   * <p>
+   * The rates provider binds the lookup to the market data.
+   * 
+   * @param marketData  the complete set of market data for one scenario
+   * @return the volatilities
+   */
+  public abstract SwaptionVolatilities volatilities(IborIndex index, MarketData marketData);
```

Comment:
Why is this method here? It would seem more natural if it were on `SwaptionMarketView`